### PR TITLE
Add new star rendering method

### PR DESCRIPTION
**Pull request by Schneegans**
This adds a new star rendering method which scales the rendered discs based on the star's magnitude. This makes the star figures more easy to spot when HDR rendering is disabled. It also improves the billboard rendering method by an updated star texture and a removed magic number.

### CODE REVIEW

File: plugins/csp-stars/src/Shaders.cpp [hunk 1]
```diff
@@ -122,11 +122,14 @@ void main() {
     float diameter = 2 * sqrt(1 - pow(1-uSolidAngle/(2*PI), 2.0));
     float scale = dist * diameter;
 
+    #ifdef DRAWMODE_SCALED_DISC
+        scale *= 1.0 - (iMagnitude - uMinMagnitude) / (uMaxMagnitude - uMinMagnitude);
+        scale *= 3.0;
```
Comment (JonasGilg) [inline plugins/csp-stars/src/Shaders.cpp:127]: Why is the scale multiplied by exactly 3?

File: plugins/csp-stars/src/Shaders.cpp [hunk 2]
```diff
@@ -122,11 +122,14 @@ void main() {
...
+    #endif
+
     #ifdef DRAWMODE_SPRITE
-        float referenceLuminance = magnitudeToLuminance(10, uSolidAngle);
-        float luminance = magnitudeToLuminance(iMagnitude, uSolidAngle);
-        float scaleFac = pow(luminance / referenceLuminance, 1.0 / 3.0);
-        scale *= scaleFac;
+        scale *= 1.0 - (iMagnitude - uMinMagnitude) / (uMaxMagnitude - uMinMagnitude);
+        scale *= 10.0;
```
Comment (JonasGilg) [inline plugins/csp-stars/src/Shaders.cpp:132]: Why is the scale multiplied by exactly 10?

File: plugins/csp-stars/src/Shaders.cpp [hunk 3]
```diff
@@ -181,13 +186,20 @@ void main() {
         float fac = luminance * clamp(1-dist, 0, 1) * 3;
     #endif
     
+    #ifdef DRAWMODE_SCALED_DISC
+        float scaleFac = 1.0 - (iMagnitude - uMinMagnitude) / (uMaxMagnitude - uMinMagnitude);
+        scaleFac *= 3.0;
```
Comment (JonasGilg) [inline plugins/csp-stars/src/Shaders.cpp:191]: Why is the scale multiplied by exactly 3?

File: plugins/csp-stars/src/Shaders.cpp [hunk 4]
```diff
@@ -181,13 +186,20 @@ void main() {
...
+
     #ifdef DRAWMODE_SPRITE
-        float referenceLuminance = magnitudeToLuminance(10, uSolidAngle);
-        float scaleFac = pow(luminance / referenceLuminance, 1.0 / 3.0);
-        // TODO: Some magic numbers here. 001744984 is the average brightness of the currently used
-        // star texture, 4.5e-12 has been retrieved by trial and error when trying to adjust the
-        // total scene brightness to the other implemented methods
-        float fac = texture(iTexture, iTexcoords * 0.5 + 0.5).r * 4.5e-12 / 0.001744984 * scaleFac / uSolidAngle;
+        float scaleFac = 1.0 - (iMagnitude - uMinMagnitude) / (uMaxMagnitude - uMinMagnitude);
+        scaleFac *= 10.0;
```
Comment (JonasGilg) [inline plugins/csp-stars/src/Shaders.cpp:197]: Why is the scale multiplied by exactly 10?

**Review by JonasGilg — COMMENTED** [review]
Appart from the comments, CI should work again, before I can approve.

**Comment by Schneegans** [comment]
I merged develop (so CI should hopefully work again) and added  some comments explaining the magic numbers.

**Review by JonasGilg — APPROVED** [review]

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**